agent/utils/common/sqlite.go

```diff
@@ -0,0 +1,86 @@
+package common
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"github.com/1Panel-dev/1Panel/agent/global"
+	"github.com/glebarez/sqlite"
+	"gorm.io/gorm"
+	"gorm.io/gorm/logger"
+)
+
+func LoadDBConnByPath(fullPath, dbName string) *gorm.DB {
+	if _, err := CreateDirWhenNotExist(true, global.CONF.System.DbPath); err != nil {
+		panic(fmt.Errorf("init db dir failed, err: %v", err))
+	}
+	if _, err := os.Stat(fullPath); err != nil {
+		f, err := os.Create(fullPath)
+		if err != nil {
+			panic(fmt.Errorf("init %s db file failed, err: %v", dbName, err))
+		}
+		_ = f.Close()
+	}
+
+	db, err := GetDBWithPath(fullPath)
+	if err != nil {
+		panic(err)
+	}
+	return db
+}
+
+func LoadDBConnByPathWithErr(fullPath, dbName string) (*gorm.DB, error) {
+	if _, err := CreateDirWhenNotExist(true, global.CONF.System.DbPath); err != nil {
+		return nil, fmt.Errorf("init db dir failed, err: %v", err)
+	}
+	if _, err := os.Stat(fullPath); err != nil {
+		f, err := os.Create(fullPath)
+		if err != nil {
+			return nil, fmt.Errorf("init %s db file failed, err: %v", dbName, err)
+		}
+		_ = f.Close()
+	}
+
+	db, err := GetDBWithPath(fullPath)
+	if err != nil {
+		return nil, fmt.Errorf("init %s db failed, err: %v", dbName, err)
+	}
+	return db, nil
+}
+
+func CloseDB(db *gorm.DB) {
+	sqlDB, err := db.DB()
+	if err != nil {
+		return
+	}
+	_ = sqlDB.Close()
+}
+
+func GetDBWithPath(dbPath string) (*gorm.DB, error) {
+	db, _ := gorm.Open(sqlite.Open(dbPath), &gorm.Config{
+		DisableForeignKeyConstraintWhenMigrating: true,
+		Logger:                                   newLogger(),
+	})
+	sqlDB, dbError := db.DB()
+	if dbError != nil {
+		return nil, dbError
+	}
+	sqlDB.SetConnMaxIdleTime(10)
+	sqlDB.SetMaxOpenConns(100)
+	sqlDB.SetConnMaxLifetime(time.Hour)
+	return db, nil
+}
+
+func newLogger() logger.Interface {
+	return logger.New(
+		log.New(os.Stdout, "\r\n", log.LstdFlags),
+		logger.Config{
+			SlowThreshold:             time.Second,
+			LogLevel:                  logger.Silent,
+			IgnoreRecordNotFoundError: true,
+			Colorful:                  false,
+		},
+	)
+}
```

Comment:
The provided GoLang implementation appears to be incomplete, with gaps around line numbers 3-5 and between lines 71-80. Below are some observations:

- **Line 3: LoadDBConnByPath**
Before loading a database connection from a given path, there should be checks such as if `CreateDirWhenNotExist` is implemented properly (lines 5-6), or whether the directory exists already (line 5). Also, adding proper logging messages about these operations would enhance readability.

**Line 71:** There's an issue where the `os.Create("dbfile")`: function could fail because it may not exist yet when the file doesn't actually need to exist immediately before usage; also, it's better to use absolute paths rather than relative ones which can potentially cause issues with OS paths.

To optimize this piece of code, consider implementing the following enhancements:

1. Properly implement logic for setting up directories (`CheckFileExists()`, create directories).
2. Ensure that the `.sqlite` extension is used explicitly instead of default ('*'), so SQLLite does get loaded properly.
3. Use more meaningful message formats like "%v -%s" or "{0}" format within the `fmt.Errorf()` calls for consistency across logs.
4. Implement logging in all methods, including initialization and closing.

Given the current state, I'd recommend completing these aspects first, then refining them according to best practices regarding performance, security, robustness, efficiency etc... As you proceed on your integration project further, ensure thorough testing throughout, covering both development environment and production scenarios thoroughly. Let me know!